lib/internal/util.js

```diff
@@ -26,7 +29,8 @@ function removeColors(str) {
 }
 
 function isError(e) {
-  return objectToString(e) === '[object Error]' || e instanceof Error;
+  // An error could be an instance of Error while not being a native error.
+  return isNativeError(e) || e instanceof Error;
```

Comment:
If we are being loose and fallback to `instanceof` anyways, why can't we just remove the first check? If an error is a native error, I'd assume the second check is always true?

Reply:
Was the previous check to allow recognizing something like `{ [Symbol.toStringTag]: 'Error' }` to bypass this check? Is that something we want to break?

Reply:
I just added another test case to outline all different cases and why it is important to have both checks in place.

The original implementation had nothing to do with the `Symbol.toStringTag`. That did not even exist at that point of time but the problem was that there was no better way of detecting errors from different realms. This is now possible with the `isNativeError` check.

Reply:
eek, forgot about realms..can you leave a comment here about cross-realm support?

Reply:
Done